### PR TITLE
migrate to Fedora 36

### PR DESCRIPTION
**Pull request by dustymabe**
Also fixup a bunch of things that are conditional on Fedora Major versions.

### CODE REVIEW

File: tests/kola/networking/default-network-behavior-change/test.sh [hunk 1]
```diff
@@ -94,7 +94,7 @@ method=auto
 org.freedesktop.NetworkManager.origin=nm-initrd-generator"
 
 # EXPECTED_REALROOT_NETWORK_CFG1:
-#   - used on F35 and RHEL <= 8.5
+#   - used RHEL <= 8.5
```
Comment (bgilbert) [inline tests/kola/networking/default-network-behavior-change/test.sh]: `used on`

**Review by bgilbert — DISMISSED** [review]
LGTM!

**Review by bgilbert — DISMISSED** [review]

File: manifest-lock.aarch64.json [hunk 2]
```diff
@@ -1,1060 +1,1081 @@
 {
```
Comment (jlebon) [inline manifest-lock.aarch64.json:1]: Are these lockfile updates copied directly from `next-devel`? Would have been good to mention that in the commit message.

File: manifests/fedora-coreos-base.yaml [hunk 3]
```diff
@@ -17,11 +17,6 @@ ostree-layers:
   - overlay/14NetworkManager-plugins
   - overlay/20platform-chrony
 
-conditional-include:
```
Comment (jlebon) [inline manifests/fedora-coreos-base.yaml:20]: s/lonber/longer/ in the commit message introducing this hunk.

File: overlay.d/35coreos-iptables/usr/lib/dracut/modules.d/35coreos-iptables/coreos-enable-iptables-legacy.sh [hunk 4]
```diff
@@ -1,82 +0,0 @@
-#!/bin/bash
```
Comment (jlebon) [inline overlay.d/35coreos-iptables/usr/lib/dracut/modules.d/35coreos-iptables/coreos-enable-iptables-legacy.sh:1]: We can also delete the `iptables-nft-migration/tests/kola` directory in the commit introducing this hunk.

File: manifests/shared-workarounds.yaml [hunk 5]
```diff
@@ -2,33 +2,6 @@
 # and downstreams (i.e. Red Hat CoreOS).
 
 postprocess:
-  # Put in the fix for multipathd.socket on releases that haven't been fixed yet.
```
Comment (jlebon) [inline manifests/shared-workarounds.yaml:5]: s/multipatch/multipath/ in commit title introducing this hunk.

File: manifests/shared-workarounds.yaml [hunk 6]
```diff
@@ -11,8 +11,8 @@ postprocess:
     set -xeuo pipefail
     source /etc/os-release
     if [[ ${NAME} =~ "Fedora" ]]; then
-        # FCOS: This fix has landed in F36+
-        [ ${VERSION_ID} -le 35 ] || exit 0
+        # FCOS: This fix has landed in FCOS
```
Comment (jlebon) [inline manifests/shared-workarounds.yaml]: Can't we simplify this further to

```bash
if [[ $ID != rhcos ]]; then
  exit 0
fi
mkdir ...
```

now?

File: manifest-lock.aarch64.json (hunk 2 above)
Comment (dustymabe) [reply manifest-lock.aarch64.json:1]: added that detail to the commit message

File: manifests/fedora-coreos-base.yaml (hunk 3 above)
Comment (dustymabe) [reply manifests/fedora-coreos-base.yaml:20]: fixed in next upload

File: overlay.d/35coreos-iptables/usr/lib/dracut/modules.d/35coreos-iptables/coreos-enable-iptables-legacy.sh (hunk 4 above)
Comment (dustymabe) [reply overlay.d/35coreos-iptables/usr/lib/dracut/modules.d/35coreos-iptables/coreos-enable-iptables-legacy.sh:1]: done in next upload

File: manifests/shared-workarounds.yaml (hunk 5 above)
Comment (dustymabe) [reply manifests/shared-workarounds.yaml:5]: fixed in next upload

File: manifests/shared-workarounds.yaml (hunk 6 above)
Comment (dustymabe) [reply manifests/shared-workarounds.yaml]: updated

**Review by jlebon — APPROVED** [review]
:tada:

**Comment by dustymabe** [comment]
CI failed with:

```
Downgrade detected. This is likely unintentional. If not, you may safely ignore this error.
```

There was one downgrade in there and we'll handle it later.

**Pull request merged**